code/game/objects/structures/morgue.dm

```diff
@@ -255,42 +257,53 @@ GLOBAL_LIST_EMPTY(crematoriums)
 		return
 
 	else
-		audible_message(span_italics("You hear a roar as the crematorium activates."))
-
+		audible_message(span_italics("You hear a roar as the crematorium fires up."))
 		locked = TRUE
 		update_icon()
+		cremate_timer = addtimer(CALLBACK(src, .proc/finish_cremate, user), (breakout_time + 3 SECONDS), TIMER_STOPPABLE)
+		
+
+/obj/structure/bodycontainer/crematorium/open()
+	. = ..()
+	if(cremate_timer)
+		locked = FALSE
+		playsound(src.loc, 'sound/machines/ding.ogg', 50, 1) //you horrible people
+		deltimer(cremate_timer)
+		cremate_timer = null
+		update_icon()
 
-		for(var/mob/living/M in conts)
+/obj/structure/bodycontainer/crematorium/proc/finish_cremate(mob/user)
+	var/list/conts = GetAllContents() - src - connected
+	audible_message(span_italics("You hear a roar as the crematorium reaches its maximum temperature."))
+	for(var/mob/living/M in conts)
+		if(M.stat != DEAD)
+			M.emote("scream")
+		if(M.client)
 			if(M.stat != DEAD)
-				M.emote("scream")
-			if(M.client)
-				if(M.stat != DEAD)
-					SSachievements.unlock_achievement(/datum/achievement/cremated_alive, M.client) //they are in body and alive, give achievement
-				SSachievements.unlock_achievement(/datum/achievement/cremated, M.client) //they are in body, but dead, they can have one achievement
-			else if(M.oobe_client) //they might be ghosted if they are dead, we'll allow it.
-				SSachievements.unlock_achievement(/datum/achievement/cremated, M.oobe_client) //no burning alive achievement if you are ghosted though
-			if(user)
-				log_combat(user, M, "cremated")
-			else
-				M.log_message("was cremated", LOG_ATTACK)
-
-			M.death(1)
-			if(M) //some animals get automatically deleted on death.
-				M.ghostize()
-				qdel(M)
-
-		for(var/obj/O in conts) //conts defined above, ignores crematorium and tray
-			qdel(O)
-
-		if(!locate(/obj/effect/decal/cleanable/ash) in get_step(src, dir))//prevent pile-up
-			new/obj/effect/decal/cleanable/ash/crematorium(src)
-
-		sleep(3 SECONDS)
```

Comment:
You removed the sleep here, this would make it so you can instantly open it again right? Consider adding it back (but don't use sleep)

Reply:
Stupid change

Reply:
no it isn't.

before the process took 3 seconds visually, but items were deleted instantly.
Now the process takes 6 seconds visually, with the items being deleted at the end.